.github/scripts/wait-for-checks.sh

```diff
@@ -2,24 +2,25 @@
 #
 # @param $1 - PR number or URL
 # wait for a bit until pr is created, otherwise it throws an error "no checks reported on the 'odh-release/e2e-test' branch"
-set -euo pipefail
+set -euo
 
 sleep 10
 
-while [ $(gh pr checks "$1" | grep -v 'tide' | grep 'pending' | wc -m) -gt 0 ]; do
-  printf "PR checks still pending, retrying in 10 seconds...\n"
-  sleep 30 # replace with 60000
-done
+pr_has_status() {
+    local pr=$1
+    local status=$2
+    local skip=${3:-tide}
 
-if [ $(gh pr checks "$1" | grep 'fail' | wc -m) -gt 0 ]; then
-  printf "!!PR checks failed!!\n"
-  exit 1
-fi
+    gh pr checks $pr | awk -v FS=$'\t' -v status=$status "\$1 ~ /$skip/{next} \$2 == status {found=1} END {if (!found) exit 1}"
+}
 
-if [ $(gh pr checks "$1" | grep 'pass' | wc -m) -gt 0 ]; then
-  printf "!!PR checks passed!!\n"
-  exit 0
-fi
+# waiting for 5 minutes before each check as e2e can take a lot of time.
+while pr_has_status $1 pending; do
+  echo "PR checks still pending, retrying in 5 minutes..."
+  sleep 300000
+done
 
-printf "!!An unknown error occurred!!\n"
+pr_has_status $1 fail && { echo "!!PR checks failed!!"; exit 1; }
```

Comment:
I'm wondering, since ```gh pr checks``` has exist status 0 (success) when there are no failed tests and non-0 otherwise,  if it's sufficient here to just run it without extra logic? Probably, "unknown error" or "failed checks" will be clear from the log?